1st commit

danylermol opened the pull request.

danylermol committed:
1st commit

danylermol commented:
Привет.

pavel-ulan commented on src/main/java/Account.java:
```diff
@@ -0,0 +1,29 @@
+
+    public boolean checkNameToEmboss(String name) {
+        //Метод удалит пробелы в начале и конце строки, а также удалит лишние пробелы внутри строки, оставив только один пробел
+        name = StringUtils.normalizeSpace(name);
```
Похвально. Использование утилит для таких задач, это клево, молодец.
Но в задании сказано проверить, а не исправить возможные ошибки, а затем проверить. Так что придется отказаться от этой нормализации и добавить проверки.

pavel-ulan commented on src/main/java/Account.java:
```diff
@@ -0,0 +1,29 @@
+    public boolean checkNameToEmboss(String name) {
+        //Метод удалит пробелы в начале и конце строки, а также удалит лишние пробелы внутри строки, оставив только один пробел
+        name = StringUtils.normalizeSpace(name);
+        name = name.toUpperCase();
```
аналогично

pavel-ulan replied:
хотя в данном случае, ты ничем с условиями не противоречишь, так что можешь и оставить.

pavel-ulan commented on src/test/java/StringLengthTest.java:
```diff
@@ -0,0 +1,70 @@
+        //Всего 45 символов в строке, после работы метода должно быть 2 символа
+        Account account = new Account("                        ab                   ");
+        Assert.assertFalse("В строке не может быть менее трёх символов", account.checkNameToEmboss(account.getName()));
+    }
```
после первого теста, этот уже избыточен

pavel-ulan reviewed (commented):
Неплохо, но есть проблема с изначальными условиями задачи и твоей интерпретацией. =)
Посмотри по замечаниям и исправь пожалуйста. Потом перепроверь соответственно класс с тестами (придется добавить проверки на пробелы как минимум). Ну и обнови скриншот со свежим прогоном после всех исправлений.
Ну и касательно папки target в репозитории. Как правило ее не должно быть (ну кроме ситуации с аллюр отчетом в задании), у тебя она тут лишняя.

danylermol committed:
-Папка target убрана;
-Тесты на длину берут по одному значению в каждом КЭ
-Добавлены проверки на пробелы (начало, не более одного в середине, конец строки)
-Тестовая строка изменена на более читаемую
-Новый метод похож на рабочий костыль сомнительного качества:)

danylermol committed:
Create jenkins

danylermol committed:
Add files via upload

pavel-ulan commented on src/main/java/Account.java:
```diff
@@ -0,0 +1,28 @@
+        String trimName = name.trim();
+        int indexOfFirstMiddleSpace = trimName.indexOf(" ", 0);
+        Character spaceChar = ' ';
+        boolean isSpaceAfterSpace = spaceChar.equals(nameArray[indexOfFirstMiddleSpace + 1]);
```
Несколько замудрено, но в целом - ок. (хотя если не лень, подумай как это можно сделать короче, т.к. сейчас это слишком сложно и по памяти и по скорости. А при решении любой задачи, всегда учитывается в том числе и качество решения, а не только решена ли задача в целом).

pavel-ulan reviewed (approved):
Хорошо. Подумай как оптимизировать проверку на пробелы в строке (просто для себя), в остальном - отлично.
